state/txindex/indexer_service.go

```diff
@@ -14,13 +17,23 @@ const (
 	subscriber = "IndexerService"
 )
 
+var (
+	LastTxIndexerRetainHeightKey    = []byte("lastTxIndexerRetainHeight")
+	LastBlockIndexerRetainHeightKey = []byte("lastBlockIndexerRetainHeight")
+	IndexerRetainHeightKey          = []byte("indexerRetainHeight")
+
+	ErrKeyNotFound        = errors.New("key not found")
+	ErrInvalidHeightValue = errors.New("invalid height value")
+)
+
 // IndexerService connects event bus, transaction and block indexers together in
 // order to index transactions and blocks coming from the event bus.
 type IndexerService struct {
 	service.BaseService
 
 	txIdxr           TxIndexer
 	blockIdxr        indexer.BlockIndexer
+	indexerStore     dbm.DB
```

Comment:
I'd like to understand a bit more about the reasoning here in keeping this `indexerStore` within this `IndexerService`. Does it not make more sense to implement this as part of the `TxIndexer` and `BlockIndexer` implementations?

Reply:
Now it is implemented as part of `TxIndexer` and `BlockIndexer`. `IndexerService` is untouched